examples/gdp/ConstrainedLayout.py

```diff
@@ -0,0 +1,151 @@
+    return model.CenterYCoord[1] - model.Radius[1] + model.Height[i]/2
+model.YCoordLB = Param(model.BOXES, initialize=min_y_coord_init)
+
+# Cost associated with the distance between box i and j
+model.DistCost = Param(model.BoxPairs, default=0)
+
+model.BoxRelations = Set(initialize=['LeftOf', 'RightOf', 'Above', 'Below'])
+
+
+## Variables #########
+
+def get_x_bounds(model, i):
+    return (model.XCoordLB[i], model.XCoordUB[i])
+model.xCoord = Var(model.BOXES, bounds=get_x_bounds)
+def get_y_bounds(model, i):
+    return (model.YCoordLB[i], model.YCoordUB[i])
+model.yCoord = Var(model.BOXES, bounds=get_y_bounds)
+
+model.horizDist = Var(model.BoxPairs, within=NonNegativeReals)
+model.vertDist = Var(model.BoxPairs, within=NonNegativeReals)
+
+
+## Constraints #########
+
+# Objective: Minimize relative distance
+def rel_dist_rule(model):
+    return sum(model.DistCost[i,j] * (model.horizDist[i,j] + model.vertDist[i,j])
+               for (i,j) in model.BoxPairs)
+model.rel_dist = Objective(rule=rel_dist_rule)
+
+
+def horiz_dist_rule1(model, i, j):
+    return model.horizDist[i,j] >= model.xCoord[i] - model.xCoord[j]
+model.horiz_dist1 = Constraint(model.BoxPairs, rule=horiz_dist_rule1)
+
+def horiz_dist_rule2(model, i, j):
+    return model.horizDist[i,j] >= model.xCoord[j] - model.xCoord[i]
+model.horiz_dist2 = Constraint(model.BoxPairs, rule=horiz_dist_rule2)
+
+def vert_dist_rule1(model, i, j):
+    return model.vertDist[i,j] >= model.yCoord[i] - model.yCoord[j]
+model.vert_dist1 = Constraint(model.BoxPairs, rule=vert_dist_rule1)
+
+def vert_dist_rule2(model, i, j):
+    return model.vertDist[i,j] >= model.yCoord[j] - model.yCoord[i]
+model.vert_dist2 = Constraint(model.BoxPairs, rule=vert_dist_rule2)
+
+
+def no_overlap_disjunct_rule(disjunct, i, j, boxRelation):
+    model = disjunct.model()   
+    if boxRelation == 'LeftOf':
+        disjunct.c = Constraint(expr=model.xCoord[i] + model.Length[i]/2 <= \
+                                model.xCoord[j] - model.Length[j]/2)
+    elif boxRelation == 'RightOf':
+        disjunct.c = Constraint(expr=model.xCoord[i] - model.Length[i]/2 >= \
+                                model.xCoord[j] + model.Length[j]/2)
+    elif boxRelation == 'Above':
+        disjunct.c = Constraint(expr=model.yCoord[i] - model.Height[i]/2 >= \
+                                model.yCoord[j] + model.Height[j]/2)
+    elif boxRelation == 'Below':
+        disjunct.c = Constraint(expr=model.yCoord[i] + model.Height[i]/2 <= \
+                                model.yCoord[j] - model.Height[j]/2)
+    else:
+        raise RuntimeError("Unrecognized box relationship: %s" % boxRelation)
+model.no_overlap_disjunct = Disjunct(model.BoxPairs, model.BoxRelations,
+    rule=no_overlap_disjunct_rule)
+
+def no_overlap_rule(model, i, j):
+    return [model.no_overlap_disjunct[i, j, direction] for direction in \
+            model.BoxRelations]
+model.no_overlap_disjunction = Disjunction(model.BoxPairs, rule=no_overlap_rule)
```

Comment:
Are we going to support the shortcut notation?
```
@model.Disjunction(model.BoxPairs)
def no_overlap_rule(model, i, j):
    ...
```

Similarly with Disjunct.

Reply:
Yes.  That notation should (already) be supported.